internal/controller/eventing/controller.go

```diff
@@ -376,8 +425,21 @@ func (r *Reconciler) handleBackendSwitching(
 }
 
 func (r *Reconciler) reconcileNATSBackend(ctx context.Context, eventing *eventingv1alpha1.Eventing, log *zap.SugaredLogger) (ctrl.Result, error) {
+	_, err := r.kubeClient.GetCRD(ctx, k8s.NatsGVK.GroupResource().String())
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			err = fmt.Errorf("NATS module has to be installed: %v", err)
+			return ctrl.Result{}, r.syncStatusWithNATSErr(ctx, eventing, err, log)
+		}
+		return ctrl.Result{}, err
+	}
+
+	if err = r.startNatsCRWatch(eventing); err != nil {
```

Comment:
Whether or not the NATS CR exists is checked after starting the watch. Could this be a problem?

Reply:
we should do it before starting watch, otherwise watch logs error in the log. It is better to watch resource if it exists.